egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,299 @@
+    parser.add_argument('--prefix', type=str, default = None, help='prefix for the id of the corrupted utterances')
+    parser.add_argument("--speech-rvb-probability", type=float, default = 0.8,
+                        help="Probability of reverberating the speech signal, e.g. 0 <= p <= 1")
+    parser.add_argument("--noise-adding-probability", type=float, default = 0.4,
+                        help="Probability of adding point-source noises, e.g. 0 <= p <= 1")
+    parser.add_argument("--max-noises-added", type=int, default = 2,
+                        help="Maximum number of point-source noises could be added")
+    parser.add_argument("input_dir",
+                        help="Input data directory")
+    parser.add_argument("output_dir",
+                        help="Output data directory")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    ## Check arguments.
+    if args.rir_list_file is None:
+        raise Exception("Rir information file must be provided")
+    
+    if not os.path.isfile(args.rir_list_file):
+        raise Exception(args.rir_list_file + "not found")
+    
+    if args.noise_list_file is not None:
+        if not os.path.isfile(args.noise_list_file):
+            raise Exception(args.noise_list_file + "not found")
+
+    if args.num_replica > 1 and args.prefix is None:
+        args.prefix = "rvb"
+        warnings.warn("--prefix is set to 'rvb' as --num-replications is larger than 1.")
+
+    return args
+
+def ParseFileToDict(file, assert2fields = False, value_processor = None):
+    if value_processor is None:
+        value_processor = lambda x: x[0]
+
+    dict = {}
+    for line in open(file, 'r'):
+        parts = line.split()
+        if assert2fields:
+            assert(len(parts) == 2)
+
+        dict[parts[0]] = value_processor(parts[1:])
+    return dict
+
+
+# This is the major function to generate pipeline command for the corruption
+# The rir list would have the following format:
+# --rir-id <string,compulsary> --room-id <string,compulsary> --receiver-position-id <string,optional> --source-position-id <string,optional> --rt-60 < <float,optional> --drr <float, optional> < location(support Kaldi IO strings) >
+# The noise list would have the following format:
+# --noise-id <string,compulsary> --noise-type <choices = (isotropic, point source),compulsary> --bg-fg-type <choices=(background|foreground), default=background> --rir-file <str, compulsary if isotropic, should not be specified if point-source> < location=(support Kaldi IO strings) >
+def CorruptWav(wav_scp, durations, output_dir, room_list, noise_list, snr_string, num_replica, prefix, speech_rvb_probability, noise_adding_probability, max_noises_added):
+    rooms = list_cyclic_iterator(room_list, random_seed = 1)
```

Comment:
I have been using a consistent default value in the python scripts I wrote, but I haven't done a check of all the scripts. I will create an issue requesting for a review of existing python scripts. We would also have to take care of the inline python scripts. 

It seems to depend on the OS.

random.seed([x])
Initialize the basic random number generator. Optional argument x can be any hashable object. If x is omitted or None, current system time is used;** current system time is also used to initialize the generator when the module is first imported. If randomness sources are provided by the operating system, they are used instead of the system time (see the os.urandom() function for details on availability).** [Reference](https://docs.python.org/2/library/random.html)